shell/browser/osr/osr_video_consumer.cc

```diff
@@ -43,15 +28,27 @@ OffScreenVideoConsumer::OffScreenVideoConsumer(
   video_capturer_->SetMinSizeChangePeriod(base::TimeDelta());
   video_capturer_->SetFormat(media::PIXEL_FORMAT_ARGB);
 
-  SizeChanged(view_->SizeInPixels());
+  // Disable any constraint to make the capture_size keep same with source_size,
+  // which will not have a letterbox no matter how size changed. It can also
+  // prevent the delay that caused by comparing content_rect with view's size
+  // when received a new frame and reset the resolution change then request a
+  // new frame.
```

Comment:
I don't understand this comment about preventing delay.

Reply:
Previously when the view's size changed (the windows dragged, maximize, etc..) it will set the new size in the `SizeChanged` and also, `SetResolutionConstraints` to limit the video capturer limit the exactly size to the view's size. 

However, there's a bug because of that. When the windows's size changes fastly, the video capturer will give out unexpected size and padding textures. It even resize the frame because of that, causing the rendered frame to be blurry. The reason is the **delay** introduced by this constraint, because the video capturer may not use the constraint until next frame (or maybe more), it may resize the frame to fit the old constraint which is not the expected behavior (lossless frames).

Then we figure out that after removing this constraint, it will still give out correct frames and doesn't affect the behavior. Actually, the resolution constriant shouldn't be used like it used to be, it don't work well with frequent changes, it should be used when you want to limit to 1080P for whole capture lifetime.

Reply:
Comments also updated :)